Added mesh debug tool - node admin page finished (#3)
ScanReach/ScanReach.Documentation: docs/ConnectPOB/manage-nodes.md

| ![Node is possible to place in map or change placement](../icons/place-node-in-map.svg) | Indicates that a node is moveable in the map. Activated by [node manager](#node-manager) |

### Showing node parent

Shows a line from the selected node to its parent. It will also show the path if there is grand parents, great grand parents etc. until it gets to the root node which is communicating with the gateway. It is a good tool to see what nodes are depending on others to be able to perform their task.

<div class="img-container">
<img src="./node-admin-images/map-images/map-parents.png" alt="Showing node with node links to neighbouring nodes and signal strength"/>
</div>

### Showing node children

Shows a line to the children of the node. A node can have several children.

<div class="img-container">
<img src="./node-admin-images/map-images/map-children.png" alt="Showing node with node links to children"/>
</div>

### Showing node neighbours

Shows a line to the 10 neighbours with best signal of a node and the signal strength to each node. These nodes are the possible parents or children of the selected node.

> 

Reviewer: ```suggestion

<div class="img-container">
<img src="./node-admin-images/map-images/map-neighbours.png" alt="Showing node with node links to neighbours"/>
</div>

```

> |     I     | Shows number of personell that is registered as embarked with wearables in the ScanReach system.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | ![POB count](./node-admin-images/toolbar-images/pob-count.png)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |

Reviewer: Fixed some typos and shortened some text.

Would also be nicer if we extracted the icons from Figma without the backgrounds?
```suggestion
| List item | Explanation                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | Variations                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
| :-------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
|     A     | Search for nodes by mac address. The number of nodes visible in the map (GA - General Arrangement) and the [node list](#node-list) will decrease as a user types the in the search bar.                                                                                                                                                                                                                                                                                                                                                                | NIL                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
|     B     | Filter toggle for placed nodes in the map.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | <ul><li class="list-no-bullet-points">![Filtering placed nodes](./node-admin-images/toolbar-images/placed-nodes-show.png)Default on - Shows nodes that is placed in the map</li><li class="list-no-bullet-points">![Filtering placed nodes](./node-admin-images/toolbar-images/placed-nodes-hide.png)Placed nodes is now hidden in map.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|     C     | Filter toggle for not placed nodes in the map.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | <ul><li class="list-no-bullet-points">![Filtering not placed nodes](./node-admin-images/toolbar-images/not-placed-nodes-show.png)Default on - Shows nodes that is NOT placed in the map</li><li class="list-no-bullet-points">![Filtering not placed nodes](./node-admin-images/toolbar-images/not-placed-nodes-hide.png)NOT placed nodes is now hidden in map.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
|     D     | Filter toggle for nodes configured as musterstations in the map. More on musterstation configuration [here](#node-manager).                                                                                                                                                                                                                                                                                                                                                                                                                            | <ul><li class="list-no-bullet-points">![Musterstation nodes](./node-admin-images/toolbar-images/musterstations-show.png)Default on - Shows nodes that is configured as musterstations in th map with the musterstation icon ![Musterstation icon](../icons/muster.svg)</li><li class="list-no-bullet-points">![Musterstation nodes](./node-admin-images/toolbar-images/musterstations-hide.png)Nodes configured as musterstations is now hidden in map.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|     E     | Filter toggle for node links in the map. More on node links [here](#showing-node-parent).                                                                                                                                                                                                                                                                                                                                                                                                                                                              | <ul><li class="list-no-bullet-points">![Node links](./node-admin-images/toolbar-images/node-links-show.png)Default on - Shows the lines between nodes depending on what is selected it shows Parent, children or neighbours</li><li class="list-no-bullet-points">![Node links](./node-admin-images/toolbar-images/node-links-hide.png)Node links is now hidden in map.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|     F     | System status. This is dependent on active system alarms and how critical these are: <ul><li>`OK` - everything is functioning well.</li><li>`Failure` - Some aspect of the system has stopped working or is not responding, this should be investigated. The system will most likely still function well.</li><li>`Critical Failure` - There is a critical error in the system that will stop core functionality of the system to work. This error should be handled immediately or contact ScanReach support for assistance in handling it.</li></ul> | <ul><li class="list-no-bullet-points">![System status ok](./node-admin-images/toolbar-images/system-status-ok.png)</li><li class="list-no-bullet-points list-with-extra-padding">![System status ok tablet](./node-admin-images/toolbar-images/system-status-ok-tablet.png)System status: OK</li><li class="list-no-bullet-points">![System status failure](./node-admin-images/toolbar-images/system-status-failure.png)</li><li class="list-no-bullet-points list-with-extra-padding">![System status failure tablet](./node-admin-images/toolbar-images/system-status-failure-tablet.png)System status: Failure</li><li class="list-no-bullet-points">![System status critical failure](./node-admin-images/toolbar-images/system-status-critical-failure.png)</li><li class="list-no-bullet-points">![System status critical failure tablet](./node-admin-images/toolbar-images/system-status-critical-failure-tablet.png)System status: Critical Failure</li></ul> |
|     G     | Mute button to mute audible alarms from the digital output node. More on the digital output node [here](#node-manager).                                                                                                                                                                                                                                                                                                                                                                                                                                | <ul><li class="list-no-bullet-points">![Mute button active](./node-admin-images/toolbar-images/mute-active.png)Possible to press button to send mute command</li><li class="list-no-bullet-points">![mute button disabled](./node-admin-images/toolbar-images/mute-disable.png)Mute button is disabled when system is sending mute command</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|     H     | Mustering toggle to initiate or deactivate Mustering in the ScanReach system. More on Mustering [here](mustering.md).                                                                                                                                                                                                                                                                                                                                                                                                                                  | <ul><li class="list-no-bullet-points">![Mustering off](./node-admin-images/toolbar-images/mustering-off.png)</li><li class="list-no-bullet-points list-with-extra-padding">![Mustering off tablet](./node-admin-images/toolbar-images/mustering-off-tablet.png)Mustering is off</li><li class="list-no-bullet-points">![Mustering on](./node-admin-images/toolbar-images/mustering-on.png)</li><li class="list-no-bullet-points">![Mustering on](./node-admin-images/toolbar-images/mustering-on-tablet.png)Mustering is initiated</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                            |
|     I     | Embarked personnel in the ScanReach system.    
| :-------: | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
|     A     | A user can search for a node by the given name of a node or the mac address of a node. The number of nodes visible in the map (GA - General Arrangement) and the [node list](#node-list) will decrease as a user types the in the search bar. Until only one node matches the searh parameter.                                                                                                                                                                                                                                                                                                                         | NIL                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
|     B     | Filtering for showing or hiding placed nodes in the map.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | <ul><li class="list-no-bullet-points">![Filtering placed nodes](./node-admin-images/toolbar-images/placed-nodes-show.png)Default on - Shows nodes that is placed in the map</li><li class="list-no-bullet-points">![Filtering placed nodes](./node-admin-images/toolbar-images/placed-nodes-hide.png)Placed nodes is now hidden in map.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|     C     | Filtering for showing or hiding not placed nodes in the map.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | <ul><li class="list-no-bullet-points">![Filtering not placed nodes](./node-admin-images/toolbar-images/not-placed-nodes-show.png)Default on - Shows nodes that is NOT placed in the map</li><li class="list-no-bullet-points">![Filtering not placed nodes](./node-admin-images/toolbar-images/not-placed-nodes-hide.png)NOT placed nodes is now hidden in map.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
|     D     | Filtering for showing or hiding nodes configured to be musterstations in the map. More on musterstation configuration [here](#node-manager).                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | <ul><li class="list-no-bullet-points">![Musterstaion nodes](./node-admin-images/toolbar-images/musterstations-show.png)Default on - Shows nodes that is configured as musterstations in th map with the musterstation icon ![Musterstation icon](../icons/muster.svg)</li><li class="list-no-bullet-points">![Musterstation nodes](./node-admin-images/toolbar-images/musterstations-hide.png)Nodes configured as musterstations is now hidden in map.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
|     E     | Filtering for showing or hiding node links in the map. These are either dotted when not selected or full drawn when selecting a node More on node links [here](#showing-node-parent).                                                                                                                                                                                                                                                                                                                                                                                                                                  | <ul><li class="list-no-bullet-points">![Node links](./node-admin-images/toolbar-images/node-links-show.png)Default on - Shows the lines between nodes dependign on what is selected it shows Parent, children or neighbours</li><li class="list-no-bullet-points">![Node links](./node-admin-images/toolbar-images/node-links-hide.png)Node links is now hidden in map.</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|     F     | Shows the system status. This is dependent on if there are system alarms and how system critical these are. <ul><li> System status: OK - everything is functioning well.</li><li> System status: Failure - Some aspect of the system has stopped working or is not responding, this should be investigated. The system will most likely still function well.</li><li> System status: Critical Failure - There is a critical error in the system that will stop core functionality of the system to work. This error should be handled immediatly or contact ScanReach support for assistance in handling it.</li></ul> | <ul><li class="list-no-bullet-points">![System status ok](./node-admin-images/toolbar-images/system-status-ok.png)</li><li class="list-no-bullet-points list-with-extra-padding">![System status ok tablet](./node-admin-images/toolbar-images/system-status-ok-tablet.png)System status: OK</li><li class="list-no-bullet-points">![System status failure](./node-admin-images/toolbar-images/system-status-failure.png)</li><li class="list-no-bullet-points list-with-extra-padding">![System status failure tablet](./node-admin-images/toolbar-images/system-status-failure-tablet.png)System status: Failure</li><li class="list-no-bullet-points">![System status critical failure](./node-admin-images/toolbar-images/system-status-critical-failure.png)</li><li class="list-no-bullet-points">![System status critical failure tablet](./node-admin-images/toolbar-images/system-status-critical-failure-tablet.png)System status: Critical Failure</li></ul> |
|     G     | Mute button to mute audible alarms from the digital output node. More on the digital output node [here](#node-manager).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | <ul><li class="list-no-bullet-points">![Mute button active](./node-admin-images/toolbar-images/mute-active.png)Possible to press button to send mute command</li><li class="list-no-bullet-points">![mute button disabled](./node-admin-images/toolbar-images/mute-disable.png)Mute button is disabled when system is sending mute command</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|     H     | A toggle to initiate or deactivate Mustering in the ScanReach system. More on Mustering [here](mustering.md).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | <ul><li class="list-no-bullet-points">![Mustering off](./node-admin-images/toolbar-images/mustering-off.png)</li><li class="list-no-bullet-points list-with-extra-padding">![Mustering off tablet](./node-admin-images/toolbar-images/mustering-off-tablet.png)Mustering is off</li><li class="list-no-bullet-points">![Mustering on](./node-admin-images/toolbar-images/mustering-on.png)</li><li class="list-no-bullet-points">![Mustering on](./node-admin-images/toolbar-images/mustering-on-tablet.png)Mustering is initiated</li></ul>                                                                                                                                                                                                                                                                                                                                                                                                                            |
|     I     | Shows number of personell that is registered as embarked with wearables in the ScanReach system.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | ![POB count](./node-admin-images/toolbar-images/pob-count.png)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
```

|                        ![Node configured as a modbus/NMEA ](../icons/dashboard.svg)                        | Indicates a node placement, configured as a modbus or NMEA node.                         |
|                ![Node configured as a modbus/NMEA with error](../icons/dashboard-alert.svg)                | Indicates a node placement, configured as a modbus or NMEA node, node has errors.        |
| ![Node is possible to place in map or change placement](./node-admin-images/map-images/node-placement.png) | Indicates that a node is moveable in the map. Activated by [node manager](#node-manager) |

### Showing node parent

Shows a line from the selected node to its parent. It will also show the path if there is grand parents, great grand parents etc. until it gets to the root node which is communicating with the gateway. It is a good tool to see what nodes are depending on others to be able to perform their task.

<div class="img-container">
<img src="./node-admin-images/map-images/map-neighbours.png" alt="Showing node with node links to neighbouring nodes and signal strength"/>

> </div>

Reviewer: Wrong image, should be node-parent

### Showing node children

Shows a line to the children of the node. A node can have several children.

<div class="img-container">
<img src="./node-admin-images/map-images/map-children.png" alt="Showing node with node links to children"/>
</div>

### Showing node neighbours

> Shows a line to all the neighbours of a node and the signal strength to each node in desibel. This is nodes that the selected node detects and could communicate with as children or parents.

Reviewer: ```suggestion
Shows a line to the 10 neighbours with best signal of a node and the signal strength to each node. These nodes are the possible parents or children of the selected node.
```

## Node list

> Shows all nodes distrubted between, Nodes located in the map and Nodes not in map.

Reviewer: ```suggestion
Shows all nodes placed in map, and not placed in map.
```

## Node manager

Opened by pressing the "Manag" button in the Node List Table Item Details [I](#node-list-table-item-details). This is where a user can place a node in the map, change location of a placed node, set a node as muster station with or without a muster box, set retention and/or sensitivity level for muster station and muster box and lastly set a node as a Alarm node.

<div class="img-container">
<img src="./node-admin-images/node-manager-images/node-manager.png" alt="Node manager"/>
</div>

### Node manager components

<div class="img-container">
<img src="./node-admin-images/node-manager-images/node-manager-open.png" alt="Node manager open"/>
</div>

## Mesh Debug Tool (MDT)

> Opened by pressing the "Debug" button in the Node List Table Item Details [(J)](#node-list-table-item-details). This is a tool for admin users or ScanReach to use during fault searching or developing. We will break down the image into sections and explain each point in the list.

Reviewer: ```suggestion
Opened by pressing the "Debug" button in the Node List Table Item Details [(J)](#node-list-table-item-details). This is a tool for admin users or ScanReach to use during debugging or internal testing. We will break down the image into sections and explain each point in the list.
```

|     D     | Time since node last reported to parent node. If time since is more than 20 minutes the time will have a orange colour.                                                                                                                                                                                                                                                                                                                    |

#### MDT - Node data 1

This is the chip in the node that handles bluetooth communication.
![Mesh Debug Tool node data section 1](node-admin-images/mesh-debug-images/mesh-debug-tool-part-2.png)

| List item | Explanation                                                                                                                                                                                                        |
| :-------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> |     A     | Section Heading                                                                                                                                                                                                    |

Reviewer: Does not need a screenshot?
Better to list the name of each item in list item column instead of A, B, C? Easier to read when you do not need to map property name to A,B,C, then look for C in the table to read description
Then the screenshot is not needed

Reviewer: So something like this:?

```MD
|    List item    | Explanation                                                                                          |
| :-------------: | :--------------------------------------------------------------------------------------------------- |
|    Udpseqno     | UDP (User Datagram Protocol) sequence number                                                         |
|     Uptime      | Uptime of node in seconds                                                                            |
|   Nodeversion   | What version is the firmware in the node running. This should be the same as Node data 1.            |
|   Bleversion    | What BLE (Bluetooth Low Energy) version is the node running. This should be the same as Node data 1. |
|  Resetattempts  | How many attempts has there been to reset the node??!?!                                              |
|  Summarycount   | Summary cout.......                                                                                  |
| Secsincesummary | Sec since summary.....                                                                               |
| Bsbuildversion  | BS build-version ......                                                                              |
| Otabuildversion | OTA build version.......                                                                             |
| Rplbuildversion | RPL build-version.......                                                                             |
```
Renders to:
![image](https://github.com/ScanReach/ScanReach.Documentation/assets/5204006/da55035b-0e21-4dad-a837-c1d6b6996aec)

|     L     | Buffers full-count.....                                                                                                                                                                                            |
|     M     | Queue full-count ......                                                                                                                                                                                            |
|     N     | Lowest Number of buffers available........                                                                                                                                                                         |
|     O     | Nordic CRC error count......                                                                                                                                                                                       |

#### MDT - Node data 2

This is the chip in the node that handles .......

> ![Mesh Debug Tool node data section 2](node-admin-images/mesh-debug-images/mesh-debug-tool-part-3.png)

Reviewer: ```suggestion
```
Same here, not really needed as we already have a screenshot of the full debug view above?

|     C     | UDP (User Datagram Protocol) sequence number                                                         |
|     D     | Uptime??                                                                                             |
|     E     | What version is the firmware in the node running. This should be the same as Node data 1.            |
|     F     | What BLE (Bluetooth Low Energy) version is the node running. This should be the same as Node data 1. |
|     G     | How many attempts has there been to reset the node??!?!                                              |
|     H     | Summary cout.......                                                                                  |
|     I     | Sec since summary.....                                                                               |
|     J     | BS build-version ......                                                                              |
|     K     | OTA build version.......                                                                             |
|     L     | RPL build-version.......                                                                             |

#### MDT - Neighbours Data

> The neighbour data is related to what other nodes this nodes "sees" around itself.
> ![Mesh Debug Tool node data section 3](node-admin-images/mesh-debug-images/mesh-debug-tool-part-4.png)
> 
> | List item | Explanation                                                                                                                                                             |
> | :-------: | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
> |     A     | Section Heading                                                                                                                                                         |
> |     B     | Time since section was updated                                                                                                                                          |
> |     C     | How many other nodes this node registeres near itself. These are nodes that this node receives signal from and could therefore be part of the data flow to the gateway. |
> |     D     | Sequencenumber.......                                                                                                                                                   |

Reviewer: ```suggestion
- Number of nodes this nodes sees around itself.

|   List item    | Explanation                                                                                                            |
| :------------: | :--------------------------------------------------------------------------------------------------------------------- |
|   Neighbours   | Number of nodes this nodes sees around itself. It is important to have 2 or more neighbours to ensure mesh redundancy. |
| Sequencenumber | Internal sequence number of the message sent to gateway                                                                |
```

#### MDT - Parent Data

> The parent data is releated to the node this node reports data to.
> ![Mesh Debug Tool node data section 4](node-admin-images/mesh-debug-images/mesh-debug-tool-part-5.png)
> 
> | List item | Explanation                    |
> | :-------: | :----------------------------- |
> |     A     | Section Heading                |
> |     B     | Time since section was updated |
> |     C     | Mac address of the parent node |

Reviewer: ```suggestion
- What node this node uses as preferred parent node.

| List item  | Explanation                    |
| :--------: | :----------------------------- |
| Parent Mac | Mac address of the parent node |
```
